GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -453,12 +453,42 @@ public override bool TryRunInstaller(InstallActionWrapper installActionWrapper,
                     return true;
                 }
             }
+        }
+
+        protected static EventMetadata CreateEventMetadata(Exception e = null)
+        {
+            EventMetadata metadata = new EventMetadata();
+            metadata.Add("Area", nameof(NuGetFeed));
+            if (e != null)
+            {
+                metadata.Add("Exception", e.ToString());
+            }
+
+            return metadata;
         }
 
         private static bool TryGetPersonalAccessToken(string gitBinaryPath, string credentialUrl, ITracer tracer, out string token, out string error)
         {
             GitProcess gitProcess = new GitProcess(gitBinaryPath, null, null);
             return gitProcess.TryGetCredentials(tracer, credentialUrl, out string username, out token, out error);
+        }
+
+        private static void ErasePersonalAccessToken(string gitBinaryPath, string credentialUrl, ITracer tracer)
+        {
+            GitProcess gitProcess = new GitProcess(gitBinaryPath, null, null);
+            gitProcess.RejectCredentials(credentialUrl);
```

Comment:
@jamill did you double check this erases using windows GCM?  It absolutely should, but I remember there is some weird logic there.

Reply:
I tested this with revoking my credentials, and GCM erased the credentials as requested.

Reply:
The version that wouldn't properly delete is prior to 1.18.4.  Credit to @mjcheetham for finding the commit https://github.com/Microsoft/Git-Credential-Manager-for-Windows/commit/d8b3b320c253cea8594fcc3f3f57d76b0abc1d79#diff-52aaad27aeab0754e3c8c6e4ee388026